backend/test/reply_comment_handler_test.py

```diff
@@ -35,30 +36,81 @@ def setUp(cls):
+        # add comment to post
+        cls.user_post.add_comment(cls.other_user_comment)
+        cls.user_post.put()
+
+
+    @patch('handlers.reply_comment_handler.send_message_notification')
+    @patch('utils.verify_token', return_value={'email': THIRD_USER_EMAIL})
+    def test_post(self, verify_token, send_message_notification):
         """Reply a comment of post"""
         # Verify size of list
-        user_comment = self.user_post.get_comment(self.comment.id)
-        self.assertEquals(len(user_comment.get('replies')), 0,
+        other_user_comment = self.user_post.get_comment(self.other_user_comment.id)
+        self.assertEquals(len(other_user_comment.get('replies')), 0,
                           "Expected size of replies list should be zero")
 
         # Call the post method
-        data_reply = {"text": "reply of comment",
-                        "institution_key": self.institution.key.urlsafe()}
-        self.testapp.post(self.URL_REPLY_COMMENT %
-            (self.user_post.key.urlsafe(), self.comment.id),
-                          json.dumps(data_reply))
+        body = {
+            'replyData': {
+                "text": "reply of comment",
+                "institution_key": self.institution.key.urlsafe()
+            },
+            'currentInstitution': {
+                'name': 'currentInstitution'
+            }
+        }
+        self.testapp.post_json(
+            URL_REPLY_COMMENT % (self.user_post.key.urlsafe(), self.other_user_comment.id),
+            body
+        )
 
         # Update post
         self.user_post = self.user_post.key.get()
 
         # Verify size of list
-        user_comment = self.user_post.get_comment(self.comment.id)
-        self.assertEquals(len(user_comment.get('replies')), 1,
-                          "Expected size of comment's list should be one")
+        other_user_comment = self.user_post.get_comment(self.other_user_comment.id)
+        self.assertEquals(
+            len(other_user_comment.get('replies')), 1,
+            "Expected size of comment's list should be one"
+        )
+
+        calls = [
+            # args used to send the notification to the post author
+            call(
+                self.user.key.urlsafe(),
+                self.third_user.key.urlsafe(),
+                "COMMENT",
+                self.user_post.key.urlsafe(),
+                body['currentInstitution']
+            ),
+            # args used to send the notification to the author 
+            # from the comment that was replyed
+            call(
+                self.other_user.key.urlsafe(),
+                self.third_user.key.urlsafe(),
+                "COMMENT",
+                self.user_post.key.urlsafe(),
+                body['currentInstitution']
+            )
+        ]
+        send_message_notification.assert_has_calls(calls)
 
         # Verify that the post is published
         self.assertEquals(self.user_post.state, "published")
```

Comment:
The tests should end here, once, test a single scope of functionality, down below, it's testing delete post, that should be another test.